ui/app/AppLayouts/Wallet/adaptors/TokenSelectorViewAdaptor.qml

```diff
@@ -0,0 +1,166 @@
+      - marketDetails: var -> object containing props like `currencyPrice` for the computed values below
+      - balances: submodel -> [ chainId:int, account:string, balance:BigIntString, iconUrl:string ]
+
+      Computed values:
+      - currencyBalance: double (e.g. `1000.42` in user's fiat currency)
+      - currencyBalanceAsString: string (e.g. "1 000,42 CZK" formatted as a string according to the user's locale)
+      - balanceAsString: string (`1.42` formatted as e.g. "1,42" in user's locale)
+    */
+
+    // input API
+    required property var assetsModel
+    required property var flatNetworksModel
+    required property string currentCurrency // CurrenciesStore.currentCurrency, e.g. "USD"
+
+    // optional filter properties; empty/default values means no filtering
+    property var enabledChainIds: []
+    property string accountAddress
+    property bool showCommunityAssets
+    property string searchString
+
+    // output model
+    readonly property SortFilterProxyModel outputAssetsModel: SortFilterProxyModel {
+        sourceModel: assetsObjectProxyModel
+
+        filters: [
+            AnyOf {
+                RegExpFilter {
+                    roleName: "name"
+                    pattern: root.searchString
+                    caseSensitivity: Qt.CaseInsensitive
+                }
+                RegExpFilter {
+                    roleName: "symbol"
+                    pattern: root.searchString
+                    caseSensitivity: Qt.CaseInsensitive
+                }
+            },
+            ValueFilter {
+                roleName: "communityId"
+                value: ""
+                enabled: !root.showCommunityAssets
+            }
+        ]
+
+        // FIXME optionally sort/filter by wallet controller as well
+        sorters: [
+            RoleSorter {
+                roleName: "currencyBalance"
+                sortOrder: Qt.DescendingOrder
+            }
+        ]
+    }
+
+    // internals
+    ObjectProxyModel {
+        id: assetsObjectProxyModel
+        sourceModel: root.assetsModel
+
+        delegate: SortFilterProxyModel {
+            id: delegateRoot
+
+            // properties exposed as roles to the top-level model
+            readonly property int decimals: model.decimals
+            readonly property double currentBalance: aggregator.value
+            readonly property double currencyBalance: {
+                if (!!model.marketDetails) {
+                    return currentBalance * model.marketDetails.currencyPrice.amount
+                }
+                return 0
+            }
+            readonly property string currencyBalanceAsString:
+                currencyBalance ? LocaleUtils.currencyAmountToLocaleString({amount: currencyBalance, symbol: root.currentCurrency,
+                                                                               displayDecimals: model.marketDetails.currencyPrice.displayDecimals})
```

Comment:
In one place `model.marketDetails` is checked (currencyBalance) and there is not. So it should be checked in both places or check skipped consistently.

Reply:
That's a recent design requirement (from yesterday's meeting) to show "nothing" instead of "0.00 USD"

Reply:
![image](https://github.com/status-im/status-desktop/assets/5377645/60481307-b697-4761-acc0-dfa6b2e25315)

Ie, "AAVE" would normally show "0.00 USD" but it should not since it doesn't have any balance for the selected account/network

Reply:
Right, fixed!